### PR TITLE
remove resource version and only port forward on pod added events

### LINKED ISSUE #3020: Spurious port-forwarding failures causing debug failure
### Actual behavior

CloudCode for VS Code is reporting frequent failures where they try to connect to a port-forwarded debug port only to have it torn down.  They see text like the following:

```
[node-guestbook-frontend-668f5947f-swbjw frontend] Debugger attached.
[node-guestbook-backend-86b575f9b6-dctsp backend] <Container was Terminated>
port forwarding pod-node-guestbook-backend-86b575f9b6-dctsp-default-8080 recovered on port 8081
port forwarding pod-node-guestbook-backend-86b575f9b6-dctsp-default-9229 recovered on port 9229
Forwarding container node-guestbook-backend-86b575f9b6-dctsp/backend to local port 8081.
Port forwarding pod/node-guestbook-backend-86b575f9b6-dctsp in namespace default, remote port 8080 -> local port 8081
Port forwarding pod/node-guestbook-backend-86b575f9b6-dctsp in namespace default, remote port 9229 -> local port 9229
failed to port forward pod-node-guestbook-backend-86b575f9b6-dctsp-default-8080, port 8081 is taken, retrying...
failed to port forward pod-node-guestbook-backend-86b575f9b6-dctsp-default-9229, port 9229 is taken, retrying...
time="2019-10-08T11:18:59-07:00" level=info msg="Streaming logs from pod: node-guestbook-backend-86b575f9b6-dctsp container: backend"

Forwarding container node-guestbook-backend-86b575f9b6-dctsp/backend to local port 8081.
Port forwarding pod/node-guestbook-backend-86b575f9b6-dctsp in namespace default, remote port 8080 -> local port 8081
Port forwarding pod/node-guestbook-backend-86b575f9b6-dctsp in namespace default, remote port 9229 -> local port 9229
[node-guestbook-backend-86b575f9b6-dctsp backend] Debugger listening on ws://127.0.0.1:9229/23df1bcb-0717-4bde-bca9-2326e3644262
[node-guestbook-backend-86b575f9b6-dctsp backend] For help see https://nodejs.org/en/docs/inspector
[node-guestbook-backend-86b575f9b6-dctsp backend] App listening on port 8080
[node-guestbook-backend-86b575f9b6-dctsp backend] Press Ctrl+C to quit.
[node-guestbook-backend-86b575f9b6-dctsp backend] connected to mongodb://node-guestbook-mongodb:27017/guestbook
[node-guestbook-frontend-7d6b988f7d-8bdcg frontend] <Container was Terminated>
Failed to launch debug session for pod 'node-guestbook-backend-86b575f9b6-dctsp', retrying 3 more time(s).
```

I've been able to reproduce this with the [nodejs-guestbook example](https://github.com/GoogleCloudPlatform/cloud-code-samples/tree/master/nodejs/nodejs-guestbook) with `skaffold dev --port-forward` with minikube:
```
Deploy complete in 484.24907ms
Port forwarding service/node-guestbook-backend in namespace default, remote port 8080 -> local port 8080
Port forwarding service/node-guestbook-mongodb in namespace default, remote port 27017 -> local port 27017
Port forwarding service/node-guestbook-frontend in namespace default, remote port 80 -> local port 4503
failed to port forward service-node-guestbook-frontend-default-80, port 4503 is taken, retrying...
Watching for changes...
[node-guestbook-backend-548c9b795-46r5f backend] <Container was Terminated>
[node-guestbook-frontend-5b88549969-f8qzq frontend] Debugger listening on ws://127.0.0.1:9229/0afdf79f-7c65-465f-aaae-86a77135945a
[node-guestbook-frontend-5b88549969-f8qzq frontend] For help see https://nodejs.org/en/docs/inspector
[node-guestbook-frontend-5b88549969-f8qzq frontend] App listening on port 8080
[node-guestbook-frontend-5b88549969-f8qzq frontend] Press Ctrl+C to quit.
[node-guestbook-backend-664c67874f-pg49x backend] Debugger listening on ws://127.0.0.1:9229/8f68e74e-0a7b-48db-9e50-d9f71cdd088d
[node-guestbook-backend-664c67874f-pg49x backend] For help see https://nodejs.org/en/docs/inspector
[node-guestbook-backend-664c67874f-pg49x backend] App listening on port 8080
[node-guestbook-backend-664c67874f-pg49x backend] Press Ctrl+C to quit.
```

CC for IntellJ are reporting that they see this issue too.  It appears to be specific to NodeJS — and it may be something specific about this example projects.

### Information

- Skaffold version: v0.39.0
- Operating system: macOS 10.14.6
- Contents of skaffold.yaml: (https://github.com/GoogleCloudPlatform/cloud-code-samples/tree/master/nodejs/nodejs-guestbook)

```yaml

apiVersion: skaffold/v1beta2
kind: Config
build:
  tagPolicy:
    sha256: {}
  # defines where to find the code at build time and where to push the resulting image
  artifacts:
  - image: node-guestbook-backend
    context: src/backend
  - image: node-guestbook-frontend
    context: src/frontend
# defines the Kubernetes manifests to deploy on each run
deploy:
  kubectl:
    manifests:
    - ./kubernetes-manifests/**.yaml
profiles:
# use the cloudbuild profile to build images using Google Cloud Build
- name: cloudbuild
  build:
    googleCloudBuild: {}
```

### Steps to reproduce the behavior

1. https://github.com/GoogleCloudPlatform/cloud-code-samples/tree/master/nodejs/nodejs-guestbook
2. `skaffold dev --port-forward` will fail sometimes, seems to happen more often with `skaffold debug --port-forward` likely because there are more ports being forwarded

### PR DESCRIPTION
**Pull request by tejal29**
Fixes #3020

**Description**

Please follow the findings [here](https://github.com/GoogleContainerTools/skaffold/issues/3020#issuecomment-542950358)
**TLDR**
1.  Portforward is unstable because, skaffold terminates the previous `kubectl port-forward` and kicks of a new `kubectl port-forward` process when a pod is modified. 
2. We don't know why a running pod's [`OwnerReferences`](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L240) field changes which causes these multiple pod Modify events. [See here](https://github.com/GoogleContainerTools/skaffold/issues/3020#issuecomment-543289323)

**Implications of this change**
1. If a user runs `kubectl edit po/podname`, and changes a container port, or container image, skaffold port-forward will be incorrect. 

However, i dont think we need to cater for this scenario.

**User facing changes**
no
**Next PRs.**

n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Mentions any output changes.
- [ ] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

Describe any user facing changes here so maintainer can include it in the release notes, or delete this block.

```
Stabilize `port-forward` by not restarting `kubectl port-forward` process for modified pods.
```

### CODE REVIEW

**Comment by balopat** [comment]
as discussed this is not the right approach - we need pod modification events to cater for situations when the port forward listener loop kicks in after the pod addition events.

**Comment by dgageot** [comment]
@tejal29 what's the status on this PR? Is is ready to review (after a rebase) or should we close it?

**Comment by tejal29** [comment]
@dgageot I will need to rebase this and again verify the binary with @sujit-kamireddy to make sure we don't see a bunch of port forward recovering logs like these due to pod being modified.
```
port forwarding pod-node-guestbook-backend-86b575f9b6-dctsp-default-8080 recovered on port 8081
port forwarding pod-node-guestbook-backend-86b575f9b6-dctsp-default-9229 recovered on port 9229
```
This will take time.

**Comment by tejal29** [comment]
I am going to close this PR for now and resurrect when i am ready to take a look at this.
